### PR TITLE
New TPV functional improvements:

### PR DESCRIPTION
**Pull request by juanjo-nan**
- Fill statement fields from journal type
- Add new buttons to sale tpv form
- Change scan domain in tpv sale

Task #156769

### CODE REVIEW

File: sale.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            quantity = None
+
+        if not quantity:
+            domain = [('salable', '=', True),
+            ['OR', ('code','=', value),
+                ('identifiers.code', '=', value),
+                ('name', 'like', '%'+value+'%'),
+                ('name', 'like', '%'+value.upper()+'%'),
+                ('name', 'like', '%'+value.capitalize()+'%'),
+                ('name', 'like', '%'+value.lower()+'%'),]]
```
Comment (albertca) [inline sale.py:32]: Why not using `ilike`?

File: sale.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        product = None
+        value = self.start.input_value
+        quantity = qty(value)
+        if len(value) > 4:
+            quantity = None
+
+        if not quantity:
+            domain = [('salable', '=', True),
+            ['OR', ('code','=', value),
```
Comment (albertca) [inline sale.py:27]: Should be idented

**Review by albertca — COMMENTED** [review]
If possible these changes should not be in this module but a customer customization one.

**Comment by juanjo-nan** [comment]
The content from this pull request has moved to https://github.com/NaN-tic/trytond-carviresa/pull/3

**Pull request closed without merging**